core/templates/dev/head/dashboard/Dashboard.js

```diff
@@ -34,10 +35,10 @@ oppia.controller('Dashboard', [
     };
 
     $rootScope.loadingMessage = 'Loading';
-    $http.get('/dashboardhandler/data').then(function(response) {
-      var data = response.data;
-      $scope.explorationsList = data.explorations_list;
-      $scope.collectionsList = data.collections_list;
+    DashboardBackendApiService.fetchDashboardData().then(function(response) {
```

Comment:
@AllanYangZhou That's a good suggestion. About displaying an error message/warning, did you meant utilising the existing functionality inside `alertsService` ?
- I don't think that we should consider response rejection as an error, a warning should be enough.
- Also noticed that there are many instances in the current codebase where we don't handle promise rejection(actually, I could find only a few places where it's handled!). I think we might need a more general solution to this. What do you think ?